org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/Argument.java

```diff
@@ -116,19 +116,25 @@ private Annotation[] getCorrespondingRecordComponentAnnotationsIfApplicable(Abst
 		return null;
 	}
 	public TypeBinding bind(MethodScope scope, TypeBinding typeBinding, boolean used) {
+		if (this.isUnnamed(scope)) {
+			scope.problemReporter().illegalUseOfUnderscoreAsAnIdentifier(this.sourceStart, this.sourceEnd, scope.compilerOptions().sourceLevel > ClassFileConstants.JDK1_8, true);
+		}
+
 		TypeBinding newTypeBinding = createBinding(scope, typeBinding); // basically a no-op if createBinding() was called before
```

Comment:
We need to check if `this` is an argument of a lambda expression ?? Otherwise we reject lambda parameters from being unnamed and report an error `As of release 21, '_' is only allowed to declare unnamed patterns, local variables, exception parameters or lambda parameters`